sled-agent/src/instance.rs

```diff
@@ -856,7 +827,14 @@ impl Instance {
         // Create a zone for the propolis instance, using the previously
         // configured VNICs.
         let zname = propolis_zone_name(inner.propolis_id());
-        let root = camino::Utf8Path::new(ZONE_ZFS_RAMDISK_DATASET_MOUNTPOINT);
+        let mut rng = rand::rngs::StdRng::from_entropy();
+        let root = inner
+            .storage
+            .all_u2_mountpoints(ZONE_DATASET)
+            .await
+            .choose(&mut rng)
+            .ok_or_else(|| Error::U2NotFound)?
+            .clone();
```

Comment:
We do this in a couple spots, but our policy is roughly the same:

- List all the U.2 "ZONE_DATASET" mountpoints
- Pick one randomly

We aren't accounting for space here, but we also delete these on reboot